GVFS/GVFS.Common/Tracing/EventListener.cs

```diff
@@ -18,24 +18,26 @@ public virtual void Dispose()
         {
         }
 
-        public void RecordMessage(string eventName, Guid activityId, Guid parentActivityId, EventLevel level, Keywords keywords, EventOpcode opcode, string jsonPayload)
+        public bool? TryRecordMessage(TraceEventMessage message, out string errorMessage)
         {
-            if (!this.IsEnabled(level, keywords))
+            if (this.IsEnabled(message.Level, message.Keywords))
             {
-                return;
+                try
+                {
+                    this.RecordMessageInternal(message);
+                }

```

Comment:
Don't we need to return `true` here?  Makes me think we don't have any testing around this code.  How hard would it be to add some unit tests?

Reply:
That's true! Thanks for spotting this. I'll add some unit tests around the `TryRecordMessage` method.